mobile_config/src/gateway_info.rs

```diff
@@ -25,23 +28,47 @@ impl GatewayInfo {
     }
 }
 
+#[derive(thiserror::Error, Debug)]
+pub enum GatewayInfoProtoParseError {
+    #[error("Invalid location string: {0}")]
+    InvalidLocation(String),
```

Comment:
```suggestion
    InvalidLocation(#[from] std::num::ParseIntError),
```
i would personally keep this context from the root error, which should include the unparseable str while still mapping it into a variant of the new unified error type you've got here and then we can drop the extra `map_err` below